git.lua

```diff
@@ -258,15 +264,29 @@ local stashes = function(token)  -- luacheck: no unused args
     -- generate matches and match filter table
     local ret = {}
     local ret_filter = {}
+    local clink_version_encoded = clink.version_encoded or 0
     for i,v in ipairs(stash_times) do
-        table.insert(ret, "stash@{"..(i-1).."}")
-        table.insert(ret_filter, "stash@{"..(i-1).."}    "..stashes[v])
+        local match = "stash@{"..(i-1).."}"
+        table.insert(ret, match)
+        if clink_version_encoded >= 10010012 then
+            table.insert(ret_filter, { match=match, type="word", description=stashes[v] })
+        elseif clink_version_encoded >= 10010009 then
```

Comment:
This is the second time this magic number appears which makes me thinking that it's worth at least moving it to a dedicated constant (along with `10010012` two lines above) but in general I'd love to see a separate module for clink version/feature detection (I think it's likely we'll see more of these version constants sooner or later)

Reply:
The format is documented in the Clink documentation.  E.g. `10020034` is v1.2.34.
The values are easily looked up in the Changes section of the documentation.

It's intended as a lightweight capability interrogation mechanism.  I'd love to have a more sophisticated system, but designing it takes careful planning.  I don't think it makes sense to block the release of Clink over that.  This lightweight mechanism covers all the bases from a functional perspective, but isn't pretty.

To respect the clink-completions code base, I can build a module for clink-completions to make version queries use friendly function names.

Reply:
Added a version module.

Reply:
I understand your concerns but I prefer to have readable and meaningful constant names rather than 'cryptic' numbers so that whoever is reading the code they don't need to look up in the changelog and can figure out what this check is about by just reading the constant name